docs/index.md

```diff
@@ -1,91 +1,121 @@
+expensive to start over. Instead Coyote can be adopted gradually, adding more and more structure
+around your _Coyote-aware_ code. The more of this structure you add the more benefit you get from
+Coyote, but it is certainly not an all or nothing proposition.
+
+So Coyote brings together elements of design, development and testing into an integrated package
+that works really well in the real world. See our [case
+studies](case-studies/azure-batch-service.md) for some great customer testimonials.
+
+## Supported programming models
+
+Coyote supports two main programming models:
+
+- [Asynchronous tasks](programming-models/async/overview.md), which follows the popular [task-based
+  asynchronous
+  pattern](https://docs.microsoft.com/en-us/dotnet/standard/asynchronous-programming-patterns/task-based-asynchronous-pattern-tap).
+  This programming model offers a `Task` type  that serves as a drop-in-replacement type for the
+  native .NET `System.Threading.Tasks.Task`. As with the native `Task`, a Coyote `Task` represents
+  an asynchronous operation that the programmer can coordinate using the `async` and `await`
+  keywords of [C#](https://docs.microsoft.com/en-gb/dotnet/csharp/). In production, a Coyote `Task`
+  executes with the same semantics of a native `Task`. In fact, it is simply a thin wrapper around a
+  native `Task` object. During testing, however, is where the magic happens. Coyote controls the
+  execution of each Coyote `Task` so that it can explore various different interleavings to find
+  bugs. Alternatively, you can use the [binary rewriting](programming-models/async/rewriting.md)
+  feature of Coyote to automatically instrument your application, taking control of
+  `System.Threading.Tasks.Task` objects and related concurrency types from the Task Parallel
+  Library, without having to use Coyote's drop-in-replacement library.
+
+- [Asynchronous actors](programming-models/actors/overview.md) is an [actor-based programming
+  model](https://en.wikipedia.org/wiki/Actor_model) that allows you to express your design and
+  concurrency at a higher-level of abstraction. This programming model starts with the `Actor` type
+  that represents a long-lived, interactive asynchronous object. An actor can create new actors,
+  send events to other actors, and handle received events. This more advanced programming model is
+  ideal for cases when asynchronous tasks get too unwieldy. This programming model also provides a
+  `StateMachine` type for easy development of event-driven state-machines. A `StateMachine` is
+  simply an `Actor` with explicit `States` and event-driven state transitions.
+
+Note that you can only systematically test the above two programming models together using Coyote's
+[binary rewriting](programming-models/async/rewriting.md) to take control of
+`System.Threading.Tasks.Task` objects. Mixing the Coyote `Actor` type with the custom Coyote `Task`
```

Comment:
Let's remove the `Mixing the Coyote ... is not supported` bit, since someone can now test the two together with rewriting?

Reply:
Fixed